File: Swift/InsertionSort.swift
import Foundation
func insertionSort(_ array: [Int]) -> [Int] {
  var a = array
  for x in 1..<a.count {
    var y = x
    let temp = a[y]
    while y > 0 && temp < a[y - 1] {
      a[y] = a[y - 1]
      y -= 1
    }
    a[y] = temp
  }
  return a
}
let Arr = [9, 2, 7, 5, 4, 3, 6, 8, 1]
print("After sorting: \(insertionSort(Arr))")

File: Rust/InsertionSort.rs
/// Author - Aditya Rana

/// Generic Insertion Sort in rust
pub fn insertion_sort<T: Ord>(arr: &mut [T]) {
    for i in 1..arr.len() {
        let mut j = i;
        while j > 0 && arr[j] < arr[j - 1] {
            arr.swap(j, j - 1);
            j = j - 1;
        }
    }
}

fn main() {
    println!("Sort numbers ascending");

    let mut numbers = [4, 65, 2, -31, 0, 99, 2, 83, 782, 1];
    println!("Before: {:?}", numbers);

    insertion_sort(&mut numbers);
    println!("After:  {:?}\n", numbers);

    println!("Sort strings alphabetically");

    let mut strings = ["beach", "hotel", "airplane", "car", "house", "art"];
    println!("Before: {:?}", strings);

    insertion_sort(&mut strings);
    println!("After:  {:?}\n", strings);
}